repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -409,6 +409,8 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
                   case Success(newAdminToken) => requestHandler.getGroups(newAdminToken, validToken.userId)
                   case Failure(x) => Failure(IdentityAdminTokenException("Unable to reacquire admin token", x))
                 }
+              case _: InvalidTokenException =>
```

Comment:
Might want to chnge the exception you are throwing, because as we figured out in this case a 404 is not telling us that the token is invalid, just that they lack groups.